eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/TagTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import org.axonframework.common.Context;
+import org.axonframework.common.SimpleContext;
+import org.junit.jupiter.api.*;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class validating the {@link Tag}.
+ *
+ * @author Steven van Beelen
+ */
+class TagTest {
```

Comment:
I'd add test, to check if tag is equal by value.

Reply:
Do you mean a direct `assertEquals(tag, tag)`, or an equals check based on the `Tag#value` result?

Reply:
A direct `assertEquals(tag, tag)`.
It's provided by record itself, but if refactor it to the class it may be broken.

Reply:
Aight, done :+1: